### PR TITLE
Various coverage improvements

### PR DESCRIPTION
**Pull request by jmsfltchr**
## What is the goal of this PR?

We add coverage for:
- unary relations
- boolean attributes (material and inferred)
- complex type inference

## What are the changes implemented in this PR?

- Adds friendship agent as a unary relation
- Adds 2 boolean attributes, one inserted, the other inferred as a status variable
- Adds a complex type `unemployed-person`
- Adds a test for the number of residencies inferred

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Misc
 .DS_Store
-logs/
+logs1/
```
Comment (adammitchelldev) [inline .gitignore]: Is this change intended?

Comment (adammitchelldev) [reply .gitignore]: Perhaps this could be covered by a glob for all log type files?

Comment (jmsfltchr) [reply .gitignore]: Not intended, and yes good idea, thanks!
